Dockerfile

```diff
@@ -60,8 +60,22 @@ RUN chmod +x /app/scripts/*.sh \
   # Build api source
   && go build -v -o photoview .
 
-### Copy api and ui to production environment ###
-FROM --platform=${BUILDPLATFORM:-linux/amd64} debian:bookworm-slim AS final
+### Build dev image for UI ###
+FROM ui AS dev-ui
```

Comment:
Because of the line 29 (`RUN npm ci --omit=dev --ignore-scripts`) this won't work for the dev purposes. UI dev image should be built with `npm ci` command in the sequence, but adding another build stage to this Dockerfile, which will be executed every time on CI and never used there, looks like wasting time and resources (as docker build already runs long time on CI).

So, in this case, it looks like it is better to create a separate Dockerfile inside the `ui` repo folder and reference it for dev-image build purposes.

Reply:
Updated.

`dev-ui` and `dev-api` won't be built when building `release` image, as `release` image doesn't depend on those stages. So it doesn't waste any more time during CI.

Reply:
```sh
$ docker build -t photoview .
[+] Building 138.3s (23/23) FINISHED                                                                   docker:default
 => [internal] load build definition from Dockerfile                                                             0.0s
 => => transferring dockerfile: 3.29kB                                                                           0.0s
 => [internal] load metadata for docker.io/library/debian:bookworm-slim                                          0.0s
 => [internal] load metadata for docker.io/library/node:18                                                       0.0s
 => [internal] load metadata for docker.io/library/golang:1.22-bookworm                                          0.0s
 => [internal] load .dockerignore                                                                                0.0s
 => => transferring context: 325B                                                                                0.0s
 => [release 1/7] FROM docker.io/library/debian:bookworm-slim                                                    0.0s
 => [internal] load build context                                                                                0.2s
 => => transferring context: 49.04MB                                                                             0.2s
 => [ui 1/4] FROM docker.io/library/node:18                                                                      0.0s
 => [api 1/5] FROM docker.io/library/golang:1.22-bookworm                                                        0.0s
 => CACHED [release 2/7] COPY scripts/install_runtime_dependencies.sh /app/scripts/                              0.0s
 => [release 3/7] RUN groupadd -g 999 photoview   && useradd -r -u 999 -g photoview -m photoview   && chmod +  119.4s
 => CACHED [api 2/5] COPY scripts /app/scripts                                                                   0.0s
 => [api 3/5] COPY api /app/api                                                                                  0.6s
 => CACHED [ui 2/4] COPY ui /app/ui                                                                              0.0s
 => CACHED [ui 3/4] WORKDIR /app/ui                                                                              0.0s
 => CACHED [ui 4/4] RUN npm ci --omit=dev --ignore-scripts   && npm run build -- --base=/                        0.0s
 => [api 4/5] WORKDIR /app/api                                                                                   0.1s
 => [api 5/5] RUN chmod +x /app/scripts/*.sh   && source /app/scripts/set_compiler_env.sh   && /app/scripts/i  133.8s
 => [release 4/7] WORKDIR /home/photoview                                                                        0.4s
 => [release 5/7] COPY api/data /app/data                                                                        0.4s
 => [release 6/7] COPY --from=ui /app/ui/dist /app/ui                                                            0.3s
 => [release 7/7] COPY --from=api /app/api/photoview /app/photoview                                              0.1s 
 => exporting to image                                                                                           3.1s 
 => => exporting layers                                                                                          3.0s 
 => => writing image sha256:04f3d0fd3d87d63b49744d734010b5bce2b86c9198362091043fcdbc5eae71ce                     0.0s 
 => => naming to docker.io/library/photoview 
```

From the staging prefix in `[]`, you can see it only builds `ui`/`api`/`release`, but no `dev-ui`/`dev-api`.

Reply:
just a question: as now the `dev-ui` is an empty stage, why exactly do we need it instead of referencing the `ui` (parent) stage in the dev-compose and individual `docker run` command?

Reply:
I know it's empty. I keep it as a symmetry stage between `api` and `ui`.